nexus/db-queries/src/db/datastore/disk.rs

```diff
@@ -771,6 +786,74 @@ impl DataStore {
 
         let attach_update = DiskSetClauseForAttach::new(authz_instance.id());
 
+        let disk = self.disk_get(&opctx, authz_disk.id()).await?;
+        let resource_query = match disk {
+            Disk::Crucible(_) => disk::table.into_boxed().filter(
+                disk::dsl::disk_state.eq_any(ok_to_attach_disk_state_labels),
+            ),
+
+            // Attaching a local storage disk to the instance has to be blocked
+            // if sled reservation has occured for this instance: local storage
+            // allocation records are only created during sled reservation, and
+            // importantly a particular configuration of local storage
+            // allocations for an instance are only _validated_ during the sled
+            // reservation.
+            //
+            // The instance start saga perform sled reservation, creates the
+            // corresponding VMM record, and changes the instance's runtime
+            // state all in _separate saga nodes_.  This means that there could
```

Comment:
take it or leave it: it may perhaps be worth noting something here about _why_ this has to happen like that, so the future reader doesn't just think "well why not just change that part?". i recall that we went down that path briefly yesterday.

Reply:
Added this basically verbatim in 7680248ab0c0e96d3a98e3dabd4ce91f5d882b8a